roomserver/storage/shared/membership_updater.go

```diff
@@ -136,7 +136,7 @@ func (u *MembershipUpdater) SetToJoin(senderUserID string, eventID string, isUpd
 		}
 
 		// Look up the NID of the new join event
-		nIDs, err := u.d.eventNIDs(u.ctx, u.txn, []string{eventID})
+		nIDs, err := u.d.eventNIDs(u.ctx, u.txn, []string{eventID}, false)
```

Comment:
Please alias this, as it's hard to know what `false` means from call-sites. Especially since `false` means 'all' not 'only sent' events.
```go
var (
    NoFilter = false
    FilterUnsentOnly = true
)
```
or something.